pywbem_mock/_cimrepository.py

```diff
@@ -2485,236 +2521,295 @@ def _pull_operations_disabled(self):
+                                   MaxObjectCount,
+                                   ContinueOnError)
 
     ####################################################################
     #
     #   OpenReferenceInstancePaths server request
     #
     ####################################################################
 
-    def OpenReferenceInstancePaths(self, **params):
+    def OpenReferenceInstancePaths(self, InstanceName, ResultClass=None,
+                                   Role=None,
+                                   FilterQueryLanguage=None, FilterQuery=None,
+                                   OperationTimeout=None, ContinueOnError=None,
+                                   MaxObjectCount=None):
         # pylint: disable=invalid-name
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenReferenceInstancePaths`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
         self.validate_namespace(namespace)
-        self._validate_open_params(**params)
-        params['ObjectName'] = params['InstanceName']
-        del params['InstanceName']
-
-        result = self.ReferenceNames(**params)
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout)
 
-        objects = [] if result is None else [x[2] for x in result[0][2]]
+        instances = self.ReferenceNames(InstanceName, ResultClass=ResultClass,
+                                        Role=Role)
 
-        return self._open_response(objects, namespace,
-                                   'PullInstancePaths', **params)
+        return self._open_response(instances, namespace,
+                                   'PullInstancePaths',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
 
     ####################################################################
     #
     #   OpenReferenceInstances server request
     #
     ####################################################################
 
-    def OpenReferenceInstances(self, **params):
+    def OpenReferenceInstances(self, InstanceName,
+                               ResultClass=None, Role=None,
+                               IncludeClassOrigin=None, PropertyList=None,
+                               FilterQueryLanguage=None, FilterQuery=None,
+                               OperationTimeout=None, ContinueOnError=None,
+                               MaxObjectCount=None):
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenReferenceInstances`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
```

Comment:
See previous comment about `namespace = self.namespace`. Only this time, the namespace is (optionally) passed in InstanceName.